backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -0,0 +1,150 @@
+		ServiceHelper.db.createUser(lockedUser, [], { loginInfo: { failedLoginCount: 10,
+			lastFailedLoginAt: new Date() } }),
+		ServiceHelper.db.createUser(lockedUserWithExpiredLock, [], { loginInfo: { failedLoginCount: 10,
+			lastFailedLoginAt: new Date('1/1/18') } }),
+		ServiceHelper.db.createUser(userWithFailedAttempts, [], { loginInfo: { failedLoginCount: 6,
+			lastFailedLoginAt: new Date() } }),
+	]);
+};
+
+const testLogin = () => {
+	describe('Login user', () => {
+		test('should log in a user using username', async () => {
+			await agent.post('/v5/login/')
+				.send({ user: testUser.user, password: testUser.password })
+				.expect(templates.ok.status);
+		});
+
+		test('should log in a user using email', async () => {
+			await agent.post('/v5/login/')
+				.send({ user: userEmail, password: testUser.password })
+				.expect(templates.ok.status);
+		});
+
+		test('should fail with an incorrect username', async () => {
+			const res = await agent.post('/v5/login/')
+				.send({ user: 'randomUsername', password: testUser.password })
+				.expect(templates.incorrectUsernameOrPassword.status);
+			expect(res.body.code).toEqual(templates.incorrectUsernameOrPassword.code);
+		});
+
+		test('should fail with an incorrect password', async () => {
+			const res = await agent.post('/v5/login/')
+				.send({ user: testUser.user, password: 'wrongPassword' })
+				.expect(templates.incorrectUsernameOrPassword.status);
+			expect(res.body.code).toEqual(templates.incorrectUsernameOrPassword.code);
+		});
+
+		test('should fail with a locked account', async () => {
+			const res = await agent.post('/v5/login/')
+				.send({ user: lockedUser.user, password: lockedUser.password })
+				.expect(templates.tooManyLoginAttempts.status);
+			expect(res.body.code).toEqual(templates.tooManyLoginAttempts.code);
+		});
+
+		test('should log in with a locked account with the lockout duration expired', async () => {
+			await agent.post('/v5/login/')
+				.send({ user: lockedUserWithExpiredLock.user, password: lockedUserWithExpiredLock.password })
+				.expect(templates.ok.status);
+		});
+
+		test('should fail with wrong password and many failed login attempts', async () => {
+			const res = await agent.post('/v5/login/')
+				.send({ user: userWithFailedAttempts.user, password: 'wrongPassword' })
+				.expect(templates.incorrectUsernameOrPassword.status);
+			expect(res.body.code).toEqual(templates.incorrectUsernameOrPassword.code);
+			expect(res.body.message).toEqual('Incorrect username or password (Remaining attempts: 3)');
+		});
+	});
+};
+
+const testLogout = () => {
+	describe('Logout user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const res = await agent.post('/v5/logout/').expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user is not logged in with API key', async () => {
```

Comment:
Or something similar.

```suggestion
		test('should fail if the user has a session via an API key', async () => {
```